### PR TITLE
Secured API routes and consolidated all api routes for /post

### PR DESCRIPTION
**Pull request by steven-tey**

### CODE REVIEW

File: pages/app/post/[id]/index.js [hunk 1]
```diff
@@ -108,11 +111,20 @@ export default function Post() {
     }
   }
 
-  const publish = async (postId) => {
```
Comment (leerob) [inline pages/app/post/[id]/index.js:111]: Should `postId` have been removed?

**Review by leerob — COMMENTED** [review]
I see a `get-post-data` still - should we move that to a GET similar to these?

File: pages/api/post.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+export default async function post(req, res) {
+  const session = await getServerSession({ req, res }, authOptions);
+  if (!session) {
+    res.status(401).end();
+    return;
+  }
+
+  switch (req.method) {
+    case "GET": {
```
Comment (NuroDev) [inline pages/api/post.js:13]: After #15 gets merged might be worth adding a HTTP methods enum to replace these

File: pages/api/post.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        },
+      });
+      const site = await prisma.site.findFirst({
+        where: {
+          id: siteId,
+        },
+      });
+      res.status(200).json({ posts, site });
+      return;
+    }
```
Comment (NuroDev) [inline pages/api/post.js:33]: Would it be better / cleaner to move each method handlers logic into unique functions?

File: pages/app/post/[id]/index.js (hunk 1 above)
Comment (steven-tey) [reply pages/app/post/[id]/index.js:111]: Yes, `postId` is a const that's defined in line 15 already so it's accessible inside the function.

File: pages/api/post.js (hunk 2 above)
Comment (steven-tey) [reply pages/api/post.js:13]: @NuroDev agreed! Love the work you're doing already, keep it up! :)

File: pages/api/post.js (hunk 3 above)
Comment (steven-tey) [reply pages/api/post.js]: Added some comments to make it clearer but we could potentially create functions for these under `/lib/post` to make it cleaner in the future too!

File: pages/api/post.js (hunk 2 above)
Comment (leerob) [reply pages/api/post.js:13]: Great idea!

**Review by leerob — APPROVED** [review]

**Pull request merged**